vars/processTemplate.groovy

```diff
@@ -1,43 +1,45 @@
 import io.openshift.Utils
 
 def call(args=[:]) {
-    def file = args.file ?: ".openshiftio/application.yaml"
-    if (!fileExists(file)) {
-        error "Application template could not be found at $file; aborting ..."
-        currentBuild.result = 'ABORTED'
-        return
-    }
+  def file = args.file ?: ".openshiftio/application.yaml"
+  if (!fileExists(file)) {
+      error "Application template could not be found at $file; aborting ..."
+      currentBuild.result = 'ABORTED'
+      return
+  }
+
 
-    def params = [:]
 
-    // sanitize input params and apply defaults
-    def inputs = args.params ?: [:]
-    inputs.each{k, v -> params[k.toUpperCase()] = v }
-    params = applyDefaults(params)
+  def templateParams = readYaml(file: file).parameters*.name ?: []
```

Comment:
Its better to `readYaml` call once instead of twice (line 20).

Reply:
`parameters*.name` could fail if no parameters are defined.

Reply:
@hrishin 

>  could fail if no parameters are defined.

Did you test it?

Reply:
```groovy
println (null*.name ?: [])
```

Reply:
@hrishin 

> Its better to readYaml call once instead of twice (line 20).

both are different. the first one is the template file and the second (line: 20) is processed output.

Reply:
ack :+1: